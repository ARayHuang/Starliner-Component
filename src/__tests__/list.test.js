import React from 'react';
import { shallow, mount } from 'enzyme';
import List from '../components/list/list';

describe('List', () => {
	it('should render correctly', () => {
		const props = {
			className: 'mock-list-className',
		};

		const wrapper = shallow(
			<List {...props}>
				<div>mock-children</div>
			</List>
		)

		expect(wrapper).toMatchSnapshot();
	});

	it('should mount in a full DOM', () => {
		const props = {
			className: 'mock-list-className',
		};
		const wrapper = mount(
			<List {...props}>
				<div>mock-children</div>
			</List>
		);

		expect(wrapper.props().className).toEqual(props.className);
		expect(wrapper.props().children).toEqual(<div>mock-children</div>);
	})
})
