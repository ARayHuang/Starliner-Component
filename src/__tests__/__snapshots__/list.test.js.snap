// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should render correctly 1`] = `
<WithStyles(ForwardRef(List))
  className="mock-list-className"
>
  <div>
    mock-children
  </div>
</WithStyles(ForwardRef(List))>
`;
